docs/upgrading.asciidoc

```diff
@@ -0,0 +1,38 @@
+[[upgrading]]
+== Upgrading
+Upgrades between minor versions of the agent, like from 1.1 to 1.2 are always backwards compatible.
+Upgrades that involve a major version bump often come with some backwards incompatible changes.
+
+Before upgrading the agent, be sure to review the:
+
+* <<release-notes,Agent release notes>>
+* {apm-overview-ref-v}/agent-server-compatibility.html[Agent and Server compatibility chart]
+
+[float]
+[[end-of-life-dates]]
+=== End of life dates
+
+We love all our products, but sometimes we must say goodbye to a release so that we can continue moving
+forward on future development and innovation.
+Our https://www.elastic.co/support/eol[End of life policy] defines how long a given release is considered supported,
+as well as how long a release is considered still in active development or maintenance.
+The table below is a simplified description of this policy.
+
+[options="header"]
+|====
+|Agent version |EOL Date |Maintained until
+|1.13.x |2021-08-11 |1.14
+|1.12.x |2021-05-21 |1.13
```

Comment:
I find this even more confusing now 😄 
EOL date is in the future (2021-05-21), yet we don't support it anymore because there's a new minor (1.13.0)

I think this table will only be really useful once we release a new Major version. In that case we can document how long the latest minor of the previous major is supported.

Reply:
I agree that this will only really be useful once a new major is released. As an example: https://github.com/elastic/apm-agent-nodejs/pull/1637/files#diff-bd9ef2ce28d1c05f980ae9ce34024b5bR38

 > EOL date is in the future (2021-05-21), yet we don't support it anymore because there's a new minor (1.13.0)

The [stack EOL table](https://www.elastic.co/support/eol) does the same thing:
7.3.x | 2021-01-31 | 7.4.0
7.4.x | 2021-04-01 | 7.5.0
7.5.x | 2021-06-02 | 7.6.0
7.6.x | 2021-08-11 | 7.7.0

Now I'm wondering if it just makes more sense to remove EOL dates altogether? Don't we really _only_ need a maintained until column?

Reply:
> I agree that this will only really be useful once a new major is released.

As we're currently trying to make creating releases more seamless I'd really appreciate it if we could introduce that table lazily, once we create a new major version. It makes sense to add the description right now of course.

Reply:
Sorry Felix, I didn't realize I never followed up on this. Updated to remove the EOL table for now.